### ISSUE TITLE
Fix race condition in Porch

### ISSUE THREAD

**Issue by liamfallon** [nephio-project/nephio#610]
Original issue URL: https://github.com/kptdev/kpt/issues/3661
Original issue user: https://github.com/mortent
Original issue created at: 2022-11-16T05:16:55Z
Original issue last updated at: 2022-11-16T05:16:55Z
Original issue body: Porch is using the go-git library from multiple threads (background poller and the requests) and it seems like the library is not thread safe. Example of how to reproduce this problem in https://github.com/GoogleContainerTools/kpt/pull/3602

Original issue comments:

**Comment by nagygergo** [comment]
Probably already fixed by https://github.com/kptdev/kpt/pull/3900

**Comment by liamfallon** [comment]
Yes, this seems to be fixed, the code mentioned in https://github.com/kptdev/kpt/pull/3602 has been refactored.
